### PR TITLE
feat: deny public network access by default

### LINKED ISSUE #186: Deny public network access by default in Azure Service Bus module
### Module Name

service-bus

### Description

Network rule set only configurable for premium SKU, however we need to ensure that when choosing premium SKU public network access is **denied** by default (as per #47), which it currently is not.

### PR DESCRIPTION
**Pull request by hknutsen**
* Update module to deny public network access by default.
* Add networking unit tests.
* Add test workflow.
* Remove redundant Go test files.
* Update development and testing instructions in README according to changes listed above.

Closes equinor/terraform-baseline#186

### CODE REVIEW

File: tests/networking.unit.tftest.hcl [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    resource_group_name        = run.setup_tests.resource_group_name
+    location                   = run.setup_tests.location
+    log_analytics_workspace_id = run.setup_tests.log_analytics_workspace_id
+  }
+
+  assert {
+    condition     = length(azurerm_servicebus_namespace.this.network_rule_set) == 0
+    error_message = "Network rule set block created when it should not have been"
+  }
+}
```
Comment (helenakallekleiv) [inline tests/networking.unit.tftest.hcl:23]: Am I understanding it correctly that we are we checking here if a network rule set block has been created, despite the `network_rule_set` list is empty?

Comment (helenakallekleiv) [reply tests/networking.unit.tftest.hcl:23]: Since its beneath the `basic_sku` test, is `network_rule_set` not availiable for basic SKU?

Comment (helenakallekleiv) [reply tests/networking.unit.tftest.hcl:23]: Same for the `standard_sku` test.

Comment (hknutsen) [reply tests/networking.unit.tftest.hcl:23]: Correct, network rules can only be configured for premium SKU, so for the basic and standard SKUs we only need to do a single check: verify that the network rule set block has **not** been created.

Comment (helenakallekleiv) [reply tests/networking.unit.tftest.hcl:23]: Alright 👍🏼

**Review by helenakallekleiv — APPROVED** [review]

**Pull request merged**